src/test/test_common.c

```diff
@@ -127,3 +127,12 @@ bool running_in_shadow() {
     // There is the same function in the Rust tests utils code
     return getenv("SHADOW_SPAWNED") != NULL;
 }
+
+bool running_in_shadow_ptrace() {
+    // There is the same function in the Rust tests utils code
+    const char* method = getenv("SHADOW_INTERPOSE_METHOD");
+    if (method == NULL) {
+        return false;
+    }
```

Comment:
nit: maybe add a `running_in_shadow()` check so that its behaviour exactly matches the rust version?

Reply:
Done